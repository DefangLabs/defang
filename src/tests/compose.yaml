name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
