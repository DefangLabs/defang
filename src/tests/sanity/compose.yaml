services:
  nginx:
    restart: unless-stopped
    image: nginx
    environment:
      - dummy
    ports:
      - target: 80
        mode: ingress
    deploy:
      resources:
        reservations:
          memory: 256M
