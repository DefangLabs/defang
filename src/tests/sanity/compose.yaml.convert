[
  {
    "name": "nginx",
    "image": "nginx",
    "platform": 2,
    "internal": true,
    "deploy": {
      "replicas": 1,
      "resources": {
        "reservations": {
          "memory": 256
        }
      }
    },
    "ports": [
      {
        "target": 80,
        "mode": 1
      }
    ],
    "secrets": [
      {
        "source": "dummy"
      }
    ],
    "networks": 1
  }
]