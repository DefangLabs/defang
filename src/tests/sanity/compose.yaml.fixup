[
  {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {
        "reservations": {
          "memory": "268435456"
        }
      },
      "placement": {}
    },
    "entrypoint": null,
    "environment": {
      "dummy": null
    },
    "image": "nginx",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 80,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ],
    "restart": "unless-stopped"
  }
]