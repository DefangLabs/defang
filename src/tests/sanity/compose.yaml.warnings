level=warning msg="service \"nginx\": ingress port without healthcheck defaults to GET / HTTP/1.1"