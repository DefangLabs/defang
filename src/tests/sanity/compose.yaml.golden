name: sanity
services:
  nginx:
    deploy:
      resources:
        reservations:
          memory: "268435456"
    environment:
      dummy: null
    image: nginx
    networks:
      default: null
    ports:
      - mode: ingress
        target: 80
    restart: unless-stopped
networks:
  default:
    name: sanity_default
